contracts/orderbook/Logic/OrderBook.sol

```diff
@@ -0,0 +1,573 @@
+            } else {
+                (tradeSize, ) = dexSwaps(getSwapAddress()).dexAmountOut(
+                    base,
+                    loanTokenAddress,
+                    collateralTokenAmount
+                );
+                if (tradeSize == 0) {
+                    return 0;
+                }
+                tradeSize = (tradeSize * leverage) / 1 ether;
+            }
+        }
+        (uint256 fSwapRate, ) = orderType == IWalletFactory.OrderType.LIMIT_OPEN
+            ? dexSwaps(getSwapAddress()).dexAmountOut(
+                loanTokenAddress,
+                base,
+                tradeSize
+            )
+            : dexSwaps(getSwapAddress()).dexAmountOut(
+                base,
+                loanTokenAddress,
+                collateralTokenAmount
+            );
+        if (fSwapRate == 0) {
+            return 0;
+        }
+        return
+            orderType == IWalletFactory.OrderType.LIMIT_OPEN
+                ? (tradeSize.TenExp(18 - int8(IERC20Metadata(loanTokenAddress).decimals())) *
+                    1 ether) /
+                    (fSwapRate.TenExp(18 - int8(IERC20Metadata(base).decimals())))
+                : (1 ether *
+                    (fSwapRate.TenExp(18 -
+                                int8(IERC20Metadata(loanTokenAddress).decimals())))) /
+                    (collateralTokenAmount.TenExp(18 - int8(IERC20Metadata(base).decimals())));
+    }
+
+    function gasPrice(address payToken) public view returns (uint256) {
+        return IPriceFeeds(getFeed()).getFastGasPrice(payToken) * 2;
+    }
+
+	function clearOrder(address trader, uint256 orderID)
+		public
+		view
+		returns (bool)
+	{
+		if(orderExpiration[trader][orderID] < block.timestamp){
+			return true;
+		}
+		uint256 swapRate = currentSwapRate(HistoricalOrders[trader][orderID].loanTokenAddress, HistoricalOrders[trader][orderID].base);
+		if(!(HistoricalOrders[trader][orderID].price > swapRate
+                ? (HistoricalOrders[trader][orderID].price - swapRate) < (HistoricalOrders[trader][orderID].price * 4) / 100
+                : (swapRate - HistoricalOrders[trader][orderID].price) < (swapRate * 4) / 100)){
+			return true;
+		}
+		return false;
+	}
+
+    function prelimCheck(address trader, uint256 orderID)
+        public
+        view
+        returns (bool)
+    {
+        if (orderExpiration[trader][orderID] < block.timestamp) {
+            return false;
+        }
+        if (
+            HistoricalOrders[trader][orderID].orderType ==
+            IWalletFactory.OrderType.LIMIT_OPEN
+        ) {
+            if (
+                HistoricalOrders[trader][orderID].loanID == 0 ||
+                isActiveLoan(HistoricalOrders[trader][orderID].loanID)
+            ) {} else {
+                return false;
+            }
+            uint256 tAmount = HistoricalOrders[trader][orderID]
+                .collateralTokenAmount > 0
+                ? HistoricalOrders[trader][orderID].collateralTokenAmount +
```

Comment:
you are reading multiple times `HistoricalOrders[trader][orderID]` from storage. load it in memory and use. this will be cleaner and gas efficient